numpy/__init__.cython-30.pxd

```diff
@@ -257,6 +257,21 @@ cdef extern from "numpy/arrayobject.h":
         NPY_MAXDIMS  # 64 on NumPy 2.x and 32 on NumPy 1.x
         NPY_RAVEL_AXIS  # Used for functions like PyArray_Mean
 
+    ctypedef struct PyDataMemAllocator:
+        void *ctx;
+        void* (*malloc) (void *ctx, size_t size);
+        void* (*calloc) (void *ctx, size_t nelem, size_t elsize);
+        void* (*realloc) (void *ctx, void *ptr, size_t new_size);
+        void (*free) (void *ctx, void *ptr, size_t size);
+
+    ctypedef struct PyDataMem_Handler:
+        char* name;  /* multiple of 64 to keep the struct aligned */
+        uint8_t version; /* currently 1 */
+        PyDataMemAllocator allocator;
+
+    object PyDataMem_SetHandler(object handler)
+    object PyDataMem_GetHandler()
```

Comment:
Nice, thanks @jakirkham I hadn't realized we never exposed it.  Can you please remove the trailing `;` and fix the comments?

Maybe also add a comment, something like:
```
# The memory handler functions require the NumPy 1.22 API
# and may require defining NPY_TARGET_VERSION
```
I think we don't have to work around that otherwise.  Because Cython should do nothing with the definition unless they are used (which should then fail to compile unfortunately with reasonable compilation settings, since an implicit function is used).

Reply:
Thanks Sebastian! 🙏

Sorry that was sloppy. Should have caught that. Updated without the C-isms ;)

Added the comment as well